api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,317 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"golang.org/x/exp/rand"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/cmdutils"
+	clientconfig "github.com/hatchet-dev/hatchet/pkg/config/client"
+	"github.com/hatchet-dev/hatchet/pkg/config/shared"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/db"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/sqlchelpers"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+func (m *MonitoringService) MonitoringPostRunProbe(ctx echo.Context, request gen.MonitoringPostRunProbeRequestObject) (gen.MonitoringPostRunProbeResponseObject, error) {
+	if !m.enabled {
+		return nil, fmt.Errorf("monitoring is not enabled")
+	}
+
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+
+	if tenant == nil || tenant.ID == "" {
+		return nil, fmt.Errorf("tenant is required")
+	}
+
+	if !slices.Contains[[]string](m.permittedTenants, tenant.ID) {
+
+		err := fmt.Errorf("tenant is not a monitoring tenant for this instance")
+
+		if len(m.permittedTenants) > 0 {
+			m.l.Error().Err(err).Msgf("monitoring tenants are %v", m.permittedTenants)
+		} else {
+			m.l.Error().Err(err).Msg("no monitoring tenants are configured")
+		}
+
+		return gen.MonitoringPostRunProbe403JSONResponse{}, err
+	}
+
+	token, err := getBearerTokenFromRequest(ctx.Request())
+
+	if err != nil {
+		return nil, err
+	}
+
+	events := make(chan string, 50)
+	errorChan := make(chan error, 50)
+	interrupt := cmdutils.InterruptChan()
+
+	cancellableContext, cancel := context.WithTimeout(ctx.Request().Context(), m.probeTimeout)
+
+	defer cancel()
+	var cleanup func() error
+
+	cf := clientconfig.ClientConfigFile{
+		Token:     token,
+		TenantId:  tenant.ID,
+		Namespace: randomNamespace(),
+		TLS: clientconfig.ClientTLSConfigFile{
+			Base: shared.TLSConfigFile{
+				TLSRootCAFile: os.Getenv("SERVER_TLS_ROOT_CA_FILE"),
```

Comment:
We shouldn't be using `os.Getenv` here because config variables are not guaranteed to be set via env. Our config loader supports loading from config files as well. It might be worth taking a look at how we initialize TLS on the server.

Reply:
The issue with the TLS is it gets processed in the loader and we don't have the raw ca cert path. Everything else just takes a config in the init so to keep that interface clean I've modified the Runtime.Monitoring config to allow a Cert file to be passed and then we check in the loader to see if it has a value and if not I use the value from the Server. 

`	if cf.Runtime.Monitoring.TLSRootCAFile == "" {
		cf.Runtime.Monitoring.TLSRootCAFile = cf.TLS.TLSRootCAFile
	}
`

Reply:
gotcha, that seems reasonable